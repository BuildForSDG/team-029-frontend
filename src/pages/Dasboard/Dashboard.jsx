import React, { useState } from 'react';
import { Sidebar as Layout, Icon } from 'semantic-ui-react';
import { Switch, Route } from 'react-router-dom';
import Sidebar from '../../components/sidebar/Sidebar.jsx';
import routes from './routes';

import './dashboard.styles.scss';

const Dashboard = () => {
	const [isSidebarOpen, setIsSidebarOpen] = useState(true);

	const handleSideBarState = (state = null) => {
		if (state) {
			return setIsSidebarOpen(state)
		}
		return setIsSidebarOpen(!isSidebarOpen);
	}

	return (
		<Layout.Pushable className="dashboard-container">
			<Sidebar isOpen={isSidebarOpen} toggleSideNav={handleSideBarState} routes={routes} />
			<Layout.Pusher className="main-content" as="div">
				<div className="top-bar">
					<div onClick={() => handleSideBarState()}>
						{!isSidebarOpen && (<Icon name="bars" className="topbar-nav-btn" />)}
					</div>
				</div>
				<div className="page-content">
					<Switch>
						{routes.map(({ main: Page, path, exact }, idx) => (
							<Route key={idx} exact={exact} path={path} render={(routeProps) => <Page {...routeProps} />} />
						))}
					</Switch>
				</div>
			</Layout.Pusher>
		</Layout.Pushable>
	);
};

export default Dashboard;
