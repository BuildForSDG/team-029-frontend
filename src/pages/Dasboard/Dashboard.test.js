import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import Dashboard from './Dashboard.jsx';

afterEach(cleanup);

it('Should render <Dashboard /> page component', () => {
  const { container, asFragment } = render(<MemoryRouter><Dashboard /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
  expect(container.querySelector('.main-content')).toBeTruthy()
  expect(container.querySelector('.topbar-nav-btn')).toBeFalsy()
});
