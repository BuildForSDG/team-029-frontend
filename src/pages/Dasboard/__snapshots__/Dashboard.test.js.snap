// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render <Dashboard /> page component 1`] = `
<DocumentFragment>
  <div
    class="pushable dashboard-container"
  >
    <div
      class="ui push left wide visible sidebar side-nav"
      icon="labeled"
      vertical="true"
    >
      <div
        class="nav-content"
      >
        <div
          class="header"
        >
          <img
            alt="logo"
            src="LOGO.svg"
          />
          <img
            alt="close"
            class="close-btn"
            src="deleteIcon.svg"
          />
        </div>
        <div
          class="user"
        >
          <div
            class="profileImg"
          >
            <img
              alt="profile"
              src="avater.jpg"
            />
          </div>
          <div>
            <h4>
              John Mark
            </h4>
            <p>
              Admin
            </p>
          </div>
        </div>
        <nav>
          <ul>
            <li>
              <div>
                <a
                  href="/dashboard"
                >
                  Dashboard
                </a>
              </div>
            </li>
            <li>
              <div>
                <a
                  href="/dashboard/worden"
                >
                  Road Wordens
                </a>
              </div>
            </li>
            <li>
              <div>
                <a
                  href="/dashboard/emergency-tip"
                >
                  Emergency Tips
                </a>
              </div>
            </li>
            <li>
              <div>
                <a
                  href="/dashboard/road"
                >
                  Road
                </a>
              </div>
            </li>
            <li>
              <div>
                <a
                  href="/dashboard/profile"
                >
                  Profile
                </a>
              </div>
            </li>
          </ul>
        </nav>
      </div>
    </div>
    <div
      class="pusher main-content"
    >
      <div
        class="top-bar"
      >
        <div />
      </div>
      <div
        class="page-content"
      />
    </div>
  </div>
</DocumentFragment>
`;
